cypress-io/cypress
# feat: warn when detecting unsupported dependencies for component testing

- Closes https://github.com/cypress-io/cypress/issues/22025

### User facing changelog


N/A

### Additional details


Add some runtime checks for CT deps, like `vite`, `react`, etc. We do these during setup. Now, we also do it each launch. This will help warn users if they've updated a dependency (like Vite) to a new version we haven't yet got official support for.

### Steps to test


You will need a project with some mismatched dependencies. This PR includes one called `component-testing-outdated-dependencies`, you could test with that, if you want. Run yarn install inside it, fire up Cypress and choose that project. You should see the appropriate warnings.

### How has the user experience changed?


Launchpad now shows a warning with the mismatched dependencies, what you should have provided, and what you did provide. TBA: the exact text, this screenshot is just showing how it looks right now. @christopherfrance will provide the correct text, as discussed [here](https://github.com/cypress-io/cypress/issues/22025#issuecomment-1193428399).

![image](https://user-images.githubusercontent.com/19196536/181448294-8bd3e6a2-c751-4ca8-b431-3940f5fc1208.png)

### PR Tasks
